scalyr_agent/agent_main.py

```diff
@@ -794,8 +928,15 @@ def __detailed_status(
         # We wait for five seconds at most to get the status.
         deadline = time.time() + 5
 
+        last_debug_stat_time = 0
         # Now loop until we see it show up.
         while True:
+            # Capture debug stats every second while waiting.
+            if debug and (last_debug_stat_time + 1 < time.time()):
```

Comment:
Per my comment above, I think it may be a good idea to allow this interval to be overridden / changed via command line flag (e.g. --stats-capture-interval 0.5 or similar).